shop/src/main/java/ro/msg/learning/shop/services/LocationService.java

```diff
@@ -0,0 +1,59 @@
+package ro.msg.learning.shop.services;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dtos.LocationDTO;
+import ro.msg.learning.shop.entities.Location;
+import ro.msg.learning.shop.entities.Stock;
+import ro.msg.learning.shop.mappers.LocationMapper;
+import ro.msg.learning.shop.repositories.LocationRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+public class LocationService {
+    private final LocationRepository locationRepository;
+    private final LocationMapper locationMapper;
+    private final EntityManager entityManager;
+
+    public List<LocationDTO> getAllLocations() {
+        List<LocationDTO> allLocationToDTO = new ArrayList<>();
+        List<Location> locations = locationRepository.findAll();
+        for (Location l : locations) {
+            allLocationToDTO.add(locationMapper.locationToLocationDTO(l));
+        }
+        return allLocationToDTO;
+    }
+
+    public List<Integer> findLocationWithProductAndQuantity(Map<Integer, Integer> productWithQuantity) {
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
```

Comment:
this can be simpliefied. The query does not need to be built here. Have a look at: https://www.baeldung.com/spring-data-jpa-query  and try and use this spring magic